sbol_utilities/helper_functions.py

```diff
@@ -277,6 +282,78 @@ def has_plasmid_role(x):
         return sbol3.SO_CIRCULAR in obj.types
     elif isinstance(obj, sbol3.SubComponent):  # if it's a subcomponent, check its definition
         return is_plasmid(find_top_level(obj.instance_of))
-        #return is_plasmid(obj.instance_of.lookup())
     else:
         return False
+
+
+class SBOL3PassiveVisitor:
```

Comment:
I like this base class a lot. It makes using the visitor pattern much easier.

Reply:
Do you want to import it up into pySBOL3, or leave it here in SBOL-utilities?

Reply:
Leave it here for now. I think we'll want to do a review in the future and decide what should get moved to pySBOL3. There are a lot of good nuggets here.